src/Chart.ts

```diff
@@ -88,9 +88,54 @@ export class Chart {
     return this;
   };
 
-  private draw = (baseElement: HTMLDivElement, bounds: Bounds) => {
+  private getXYMinMax = (points: Point[]) => {
+    const scales: Scales = {
+      x: { start: Infinity, end: -Infinity },
+      y: { start: Infinity, end: -Infinity }
+    };
+    for (let i = 0; i < points.length; i++) {
+      const { x, y } = points[i];
+      if (x < scales.x.start) scales.x.start = x;
+      if (x > scales.x.end) scales.x.end = x;
+      if (y < scales.y.start) scales.y.start = y;
+      if (y > scales.y.end) scales.y.end = y;
+    }
+    return scales;
+  };
+
+  private processScales = (partialScales: PartialScales): Scales => {
+    const traceLayers = this.optionalLayers.filter(l => l.type === LayerType.Trace) as TracesLayer[];
+    const scatterLayers = this.optionalLayers.filter(l => l.type === LayerType.Scatter) as ScatterLayer[];
+    let flatPointsDC = traceLayers.reduce((points, layer) => {
+      return [...layer.linesDC.map(l => l.points).flat(), ...points];
+    }, [] as Point[]);
+    flatPointsDC = scatterLayers.reduce((points, layer) => {
+      return [...layer.points.map(p => ({ x: p.x, y: p.y })), ...points];
+    }, flatPointsDC);
+
+    const minMax = this.getXYMinMax(flatPointsDC);
+
+    const yPaddingFactor = 0.1;
+    const yStart = partialScales.y?.start ?? minMax.y.start;
+    const yEnd = partialScales.y?.end ?? minMax.y.end;
+    const yRange = Math.abs(yStart - yEnd);
```

Comment:
It amounts to the same thing, but I feel like this is the more expected way of putting it!
```suggestion
    const yRange = Math.abs(yEnd - yStart);
```